docs/DeveloperGuide.md

```diff
@@ -164,79 +170,133 @@ recurrence it is (e.g. `DailyRecurrence`, `null`) which is encapsulated as a `Re
+
+The following are the used methods in `Recurrence`:
++ `checkIfDateAltered(...)`
 
 The following are child classes of `Recurrence`:
 + `DailyRecurrence`: Handles entries labelled as daily recurring events
 + `MonthlyRecurrence`: Handles entries labelled as monthly recurring events
 + `YearlyRecurrence`: Handles entries labelled as yearly recurring events
 
-##### RecurrenceFrequency enumeration
-The `RecurrenceFrequency` enumeration is used to determine the type of recurring entry of `EntryType` and its child
-classes <br>
-Enumeration constants:
-+ `NONE`: Represents a none recurring entry
-+ `DAILY`: Represents a daily recurring entry
-+ `MONTHLY`: Represents a monthly recurring entry 
-+ `YEARLY`: Represents a yearly recurring entry
-
-##### parseRecurrence method
-Class: `Parser` <br>
-Method Signature: <br>
-```
-public static Recurrence parseRecurrence(Type entry)
-```
-Functionality: <br>
-1. Takes in child class of `Type` (i.e. `Spending`, `Income`)
-2. Matches the `reccurenceFrequency` attribute with switch case to determine which `Recurrence` child to return
-3. Returns `DaillyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` or `null`(If not a recurring entry).
-
 ##### checkIncomeRecurrence / checkSpendingRecurrence method
 Class: `DailyRecurrence`, `MonthlyRecurrence`, `YearlyRecurrence` <br>
 Method Signature: <br>
 ```
 @Override
-public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes)
+public void checkIncomeRecurrence(Income recurringIncome, IncomeList incomes, boolean isAdding)
 @Override
-public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings)
+public void checkSpendingRecurrence(Spending recurringSpending, SpendingList spendings, boolean isAdding)
 ```
 Below illustrates the functionality of the checkIncomeRecurrence method through a sequence diagram <br>
 <br>
-<img src="./Diagrams/Recurrence/addRecurrenceEntry.png" alt="addRecurrenceEntry" width="800"/> <br>
+<img src="./Diagrams/Recurrence/addRecurrenceEntrySD.png" alt="addRecurrenceEntrySD" width="800"/> <br>
 Note that recurrence frequency is either 1 day (daily), 1 month (monthly) or 1 year (yearly). <br>
-Since checkSpendingRecurrence method follows the same sequence as checkIncomeRecurrence method, the diagram is omitted 
+Since checkSpendingRecurrence method follows the same sequence as checkIncomeRecurrence method, the diagram is omitted
 for brevity.
 
 Functionality: <br>
 1. Checks `lastRecurred` attribute of `recurringIncome`/`recurringSpending` against the current date via `LocalDate.now`
 2. According to the type of recurrence, check if enough time has passed between the 2 dates
 3. Adds additional recurring entries into the `IncomeList`/`SpendingList` if needed.
 
+Note: <br>
+Parameter `isAdding` is illustrated in the sequence diagram above in the opt condition. It allows the method to add 
+entries to list if set to `true`, it may be set to false when used for backlogging
```

Comment:
maybe the note can be moved to inside the sequence diagram instead

Reply:
alrighty